extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.support.jackson.datafromat.xml.deployment;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import javax.xml.stream.XMLEventFactory;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLOutputFactory;
+
+import com.ctc.wstx.shaded.msv.org_isorelax.verifier.VerifierFactoryLoader;
+import com.ctc.wstx.shaded.msv.relaxng_datatype.DatatypeLibraryFactory;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.ObjectCodec;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.DeploymentClassLoaderBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
+import io.quarkus.deployment.util.ServiceUtil;
+import org.codehaus.stax2.validation.XMLValidationSchemaFactory;
+
+public class JacksonDataformatXmlSupportProcessor {
+
+    private static final String FEATURE = "camel-support-jackson-dataformat-xml";
+    static final String SERVICES_PREFIX = "META-INF/services/";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    void serviceProviders(BuildProducer<ServiceProviderBuildItem> serviceProviders,
+            final DeploymentClassLoaderBuildItem classLoader) {
+        Stream.concat(
+                Stream.of(
+                        JsonFactory.class,
+                        ObjectCodec.class,
+                        VerifierFactoryLoader.class,
+                        DatatypeLibraryFactory.class,
+                        XMLEventFactory.class,
+                        XMLInputFactory.class,
+                        XMLOutputFactory.class)
+                        .map(Class::getName),
```

Comment:
maybe we should just use class names as strings

Reply:
Yeah, we discussed this some time ago with the result that strings are faster and they do not have any disadvantages related to class loading. In the meantime, I asked David about this and he said he finds the perf drawbacks negligible and so I started preferring maintainability. WDYT?

Reply:
I'm always +1 for maintainability you know :)